merlin-framework/src/main/java/gov/nasa/jpl/aerie/merlin/framework/ThreadedTask.java

```diff
@@ -43,28 +43,28 @@ public TaskStatus step(final Scheduler scheduler) {
       this.hostToTask.put(new TaskRequest.Resume(scheduler));
       final var response = this.taskToHost.take();
 
-      if (response instanceof TaskResponse.Success) {
-        final var status = ((TaskResponse.Success) response).status;
+      if (response instanceof TaskResponse.Success successResponse) {
```

Comment:
Personally, I tend to use very short names like `r` for the instanceof-introduced variables, because they tend to have very constraint scope. It also means that every branch gets the same name, which avoids introducing lots of names for what really is the same object, even as we interact with it differently in each branch.

In the future ((JEP 405)[https://openjdk.java.net/jeps/405]), we'll be able to extract the fields of this records inline in the pattern, so even this name will go away. (Something like `response instanceof TaskResponse.Success(var status)`.)